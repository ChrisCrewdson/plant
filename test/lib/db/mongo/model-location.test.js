const helper = require('../../../helper');
const mongo = require('../../../../lib/db/mongo')();
const { mockLogger } = require('../../../mock-logger');

describe('/lib/db/mongo/model-location', () => {
  /** @type {string} */
  let userId;
  let fbUser;
  // let locationId;

  beforeAll(async () => {
    const data = await helper.startServerAuthenticated();
    fbUser = data.user;
    userId = fbUser._id;
    expect(userId).toBeTruthy();
    // locationId = data.user.locationIds[0]._id;
    expect(typeof userId).toBe('string');
    Object.freeze(fbUser);
  });
  afterAll(() => helper.stopServer());

  describe('Create', () => {
    test('should fail to create a location if members and createdBy are missing', async () => {
      expect.hasAssertions();
      try {
        const loc = {};
        await mongo.createLocation(loc, mockLogger);
      } catch (e) {
        expect(e.message).toEqual('members and createdBy must be specified as part of location when creating a location');
      }
    });

    test('should fail to create a location if members is missing', async () => {
      expect.hasAssertions();
      try {
        const loc = {
          createdBy: userId,
        };
        await mongo.createLocation(loc, mockLogger);
      } catch (e) {
        expect(e.message).toEqual('members and createdBy must be specified as part of location when creating a location');
      }
    });

    test('should create a location if members and createdBy are present', async () => {
      const loc = {
        createdBy: userId,
        members: {
          [userId]: 'owner',
        },
      };
      const actual = await mongo.createLocation(loc, mockLogger);
      const expected = {
        _id: actual._id,
        createdBy: userId,
        members: {
          [userId]: 'owner',
        },
      };
      expect(actual).toEqual(expected);
    });
  });

  // describe('Read', () => {});
  // describe('Update', () => {});
  // describe('Delete', () => {});
});
