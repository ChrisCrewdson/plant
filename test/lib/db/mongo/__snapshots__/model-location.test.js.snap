// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/lib/db/mongo/model-location Create should fail to create a location if members is missing 1`] = `[Error: members, title and createdBy must be specified when creating a location]`;

exports[`/lib/db/mongo/model-location Create should fail to create a location if members, title and createdBy are missing 1`] = `[Error: members, title and createdBy must be specified when creating a location]`;
